test/unit/featureVisibility/FeatureVisibilityServiceTest.php

```diff
@@ -0,0 +1,179 @@
+                    return '';
+                }
+                return $this->config[$key];
+            }
+
+            public function set($key, $value)
+            {
+                $this->config[$key] = $value;
+            }
+        };
+
+        $registryStub = new class($abstractRegistryStub) extends ClientLibConfigRegistry {
+            private static $abstractRegistryStub;
+
+            public function __construct($abstractRegistryStub)
+            {
+                self::$abstractRegistryStub = $abstractRegistryStub;
+            }
+
+            public static function getRegistry()
+            {
+                return self::$abstractRegistryStub;
+            }
+        };
+
+        $this->abstractRegistryStub = $abstractRegistryStub;
+
+        $this->featureVisibilityService = new FeatureVisibilityService($registryStub);
+    }
+
+    public function testShowFeatureSetsFeatureToTheShowStatus()
+    {
+        $featureName = 'item1/customInteraction/*';
+
+        $this->featureVisibilityService->showFeature($featureName);
+
+        $resultConfig = $this->abstractRegistryStub->get('helpers/features');
+        $this->assertEquals(
+            FeatureVisibilityService::SHOW_PARAM,
+            $resultConfig['visibility'][$featureName]
+        );
+    }
+
+    public function testHideFeatureSetsFeatureToTheHideStatus()
+    {
+        $featureName = 'item/multiColumn';
+
+        $this->featureVisibilityService->hideFeature($featureName);
+
+        $resultConfig = $this->abstractRegistryStub->get('helpers/features');
+
+        $this->assertEquals(
+            FeatureVisibilityService::HIDE_PARAM,
+            $resultConfig['visibility'][$featureName]
+        );
+    }
+
+    public function testSetFeaturesVisibilitySetsFeaturesToTheGivenStatusesAndDoesNotRemovePreviouslySetFeatures()
+    {
+        $singleFeatureName = 'item/multiColumn';
+
+        $featuresMap = [
+            "item/customInteraction/audioPciInteraction" => FeatureVisibilityService::SHOW_PARAM,
+            "test/item/timeLimits" => FeatureVisibilityService::HIDE_PARAM,
+        ];
+
+        $this->featureVisibilityService->showFeature($singleFeatureName);
+        $this->featureVisibilityService->setFeaturesVisibility($featuresMap);
+
+        $resultConfig = $this->abstractRegistryStub->get('helpers/features');
+        $this->assertEquals(
+            $featuresMap + [$singleFeatureName => FeatureVisibilityService::SHOW_PARAM],
+            $resultConfig['visibility']
+        );
+    }
+
+    public function testSetFeaturesVisibilityThrowsExceptionInCaseOfWrongStatus()
+    {
+        $wrongFeatureStatus = 'wrongStatus';
+
+        try {
+            $this->featureVisibilityService->setFeaturesVisibility(['featureName' => $wrongFeatureStatus]);
+            $this->assertTrue(false);
+        } catch (Exception $e) {
+            $this->assertStringContainsString($wrongFeatureStatus, $e->getMessage());
+        }
```

Comment:
We can use `phpunit` API to assert the exception:

```suggestion
        $this->expectException(Exception::class);
        
        $this->featureVisibilityService->setFeaturesVisibility(['featureName' => $wrongFeatureStatus]);
```

Reply:
fixed